productivity-suite/app/fragments/login/src/components/LoginForm.tsx

```diff
@@ -49,12 +49,9 @@ export const LoginForm = component$(() => {
   });
 
   const dispatchLoginEvent = $(() => {
-    dispatchPiercingEvent(ref.value!, {
-      type: "login",
-      payload: {
-        username: state.username,
-        password: state.password,
-      },
+    getBus(ref.value!).dispatch("authentication", {
```

Comment:
The ! is not actually needed from a type checking point of view,
since getBus() can accept undefined.

Although I think that would be a bug. So the safest thing would be to check that `ref.value` is not undefined and throw an error if it is, rather than just telling TS to ignore the undefined typing.

Reply:
good point, I'm pretty sure I added the `!` when we were accepting `Element | null` so `undefined` was actually causing typing issues, and missed it during the refactoring 😅 

I agree that now we can just do `getBus(ref.value)`

I don't think that it can ever be `undefined` since we're just applying it below in the jsx (and this is not a generic/reusable code), do you totally think that a check for that is needed?

Reply:
I've removed the `!`, please let me know if you think I also need to add the check 🙂